File: Assignments/Solutions/Induja Shankar/Assignment 4/q5_anagram.cpp
#include <bits/stdc++.h>
#include <cstring>
using namespace std;

bool anagram_check(char str1[], char str2[]) {
    bool flag = false;
    if(strlen(str1) == strlen(str2)) 
    {
        for(int i = 0; i < strlen(str1); i++){
            flag = false;
            for(int j = 0; j < strlen(str1); j++) {
                if(str1[i] == str2[j]) {
                    flag = true;
                    break;
                }
            }
            if(flag == false)
                break;
        }
    }
    return flag;
}

int main()
{
    char str1[100], str2[100];
    cin >> str1 >> str2;
    
    if(anagram_check(str1, str2))
        cout << "Yes. Two strings are anagram of each other" << endl;
    else
        cout << "No. The strings are not anagrams of each other" << endl;
    
    return 0;
}

File: Assignments/Solutions/Anuj Nandeshwar/Assignment4/Q5Anagram.java
// Question 5

// Given two strings, write program to check whether two given strings are anagram of each other or not
// An anagram of a string is another string that contains the same characters, only the order of characters can be different

// I/P:
// str1="triangle" , str2="integral"
// O/P:
// Yes. Two strings are anagram of each other
import java.util.*;
public class Q5Anagram {
    public static boolean checkAnagram(String s1,String s2) {
        char[] cs1 = s1.toCharArray();
        char[] cs2 = s2.toCharArray();    
        int l1 = s1.length();
        int l2 = s2.length();
        if(l1!=l2){
            return false;
        }   
        Arrays.sort(cs1);
        Arrays.sort(cs2);
        for (int i = 0; i < l1; i++){
            if(cs1[i] != cs2[i]){
                return false;
            }
        } 
        return true;  
        
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter String 1");
        String s1 = sc.nextLine();
        System.out.println("Enter String 2");
        String s2 = sc.nextLine();
        
        if(checkAnagram(s1,s2)){
            System.out.println("Yes. Two strings are anagram of each other");
        }
        else{
            System.out.println("No. Two strings are not an anagram of each other");
        }

    }
}
